### PR TITLE
(tests) - coverage

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Decided to push my changes on a new branch after a mistake.... 

- [x] component tests
- [x] useSubscription handler test
- [x] debugExchange tests
- [ ] toSuspenseSource is missing cases for cancellation -->  figuring it out
- [x] missing teardown case for fetch
- [x] cacheExchange missing undefined data case

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
Sorry but before I continue this I'll have to familiarize myself a bit more with the concepts used in toSuspenseSource  etc

**Comment by kitten** [comment]
@JoviDeCroock No worries, this has already upped the coverage by a _lot_ :tada:

The `toSuspenseSource` extra test isn't that important anyway, since we're only using it for SSR in which case we'll never hit that cancellation branch :+1:

**Comment by JoviDeCroock** [comment]
The timing, just when I hit ready for review, you respond haha

File: src/exchanges/fetch.test.ts [hunk 1]
```diff
@@ -92,3 +92,17 @@ it('calls cancel when the Observable is cancelled', () => {
...
 });
+
+it('should not start the operation when teardown gets passed', () => {
+  pipe(
+    fromValue({
+      ...queryOperation,
+      operationName: 'teardown' as OperationType,
+    }),
+    fetchExchange(exchangeArgs),
+    subscribe(fail)
```
Comment (kitten) [inline src/exchanges/fetch.test.ts]: haven't used `fail` in a while 👌 totally forgot that existed

**Comment by JoviDeCroock** [comment]
I'm happy with it, having some issues figuring toSuspenseSource out. Have gotten a better grasp of the codebase which was my intention.

**Comment by kitten** [comment]
@JoviDeCroock Yea, no worries; `toSuspenseSource` is definitely a bit of a weird one. I'll see if I find time to write the missing tests and add some more explanations on what it's supposed to do 🙌

**Comment by kitten** [comment]
💪

Screenshot 2019-06-11 at 21 32 30

**Pull request merged**